### Scipy kütüphanesinden 'ttest_1samp()' fonksiyonunu kullanarak tek örneklem t-testi yapacağız. Bu fonksiyon, örneklem verileri ve hipotezlerimizle birlikte çalışır.

In [1]:
import numpy as np
from scipy.stats import ttest_1samp

In [9]:
sample_data = np.array([970, 990, 985, 975, 995, 980, 1005, 1000, 990, 980,
                        975, 1005, 990, 1000, 985, 980, 1000, 975, 985, 995,
                        990, 1005, 985, 980, 1000, 975, 980, 985, 990, 995,
                        990, 995, 980, 985, 980, 975, 1000, 1000, 985, 980,
                        995, 990, 975, 990, 1000, 980, 985, 990, 1000, 1000])

In [10]:
# Hipotezlerimiz:

null_hypothesis = 1000
alpha = 0.05

In [11]:
# Tek örneklem t-testi

t_statistic, p_value = ttest_1samp(sample_data, null_hypothesis)

In [12]:
# Hesaplanan t istatistiği ve p değerini yazdıralım

print('T istatistiği: ', t_statistic)
print('P değeri: ', p_value)

T istatistiği:  -8.63439002420002
P değeri:  2.0811894358680954e-11


### Yukarıda hesaplanan t istatistiği -8.634 ve p değeri 0.000000000020811894358680954 olarak bulunmuştur. Bu sonuçlar, null hipotezin reddedilmesi gerektiğini göstermektedir çünkü p değeri alfa seviyesinden daha küçüktür. Yani, ürünlerin ortalama ömrünün 1000 saatten farklı olduğunu söyleyebiliriz.